homeassistant/components/tuya/models.py

```diff
@@ -157,10 +157,35 @@ def read_device_status(self, device: CustomerDevice) -> bool | None:
 
 
 @dataclass(kw_only=True)
-class DPCodeEnumWrapper(DPCodeWrapper):
+class DPCodeTypeInformationWrapper[T: TypeInformation](DPCodeWrapper):
+    """Base DPCode wrapper with Type Information."""
+
+    dptype: DPType = None  # type: ignore[assignment]
```

Comment:
This is a bit meh, I guess it's because the class is a dataclass, right?
Why is this class hieararchy even dataclasses, maybe just implement an `__init__` method here, or do we need other dataclass features?